src/main/java/Deadline.java

```diff
@@ -0,0 +1,16 @@
+public class Deadline extends Task{
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.getDescription() + " (by: " + by + ")";
```

Comment:
maybe you should make the use of constants in order to avoid magic strings here...